### ISSUE TITLE
[Bug]: mo painc by runtime error: slice bounds out of range [:-1] when start on standlone mode 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#12232]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):352abd68988e58f150ffaed46a424bf777d39ec4
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

error log:
4967295:0}|goRoutineId 25875|37be9c50-6f13-11ee-8481-ce1bdd7cf1ea","session_id":"37be9c50-6f13-11ee-8481-ce1bdd7cf1ea"}
{"level":"ERROR","time":"2023/10/20 14:38:05.562958 +0800","name":"log-service.frontend","caller":"frontend/routine_manager.go:493","msg":"Error occurred","uuid":"7c4dccb4-4d3c-41f8-b482-5251dc7a41bf","error":"internal error: client send quit","session_info":"connectionId 104|127.0.0.1:60395|{account sys:87c9e648-1664-4652-acd0-e5b758b99568:moadmin -- 0:4294967295:0}|goRoutineId 25875|37be9c50-6f13-11ee-8481-ce1bdd7cf1ea","session_id":"37be9c50-6f13-11ee-8481-ce1bdd7cf1ea"}
{"level":"ERROR","time":"2023/10/20 14:38:05.562997 +0800","caller":"v2@v2.0.0-20230811091418-ff72333eabdc/application.go:383","msg":"session handle failed, close this session","listen-addresses":"[0.0.0.0:6001]","session-id":4,"addr":"127.0.0.1:60395","error":"internal error: client send quit"}
{"level":"ERROR","time":"2023/10/20 14:38:05.563024 +0800","caller":"v2@v2.0.0-20230811091418-ff72333eabdc/application.go:345","msg":"handle session failed","listen-addresses":"[0.0.0.0:6001]","error":"internal error: client send quit"}
{"level":"INFO","time":"2023/10/20 14:38:06.540169 +0800","caller":"blockio/pipeline.go:468","msg":"SelectivityStats: BLK[0/0=0.0000] COL[0/0=0.0000] RDF[0/0=0.0000,0/0=0.0000]RDD[0s/0s/0s/0]"}
{"level":"INFO","time":"2023/10/20 14:38:06.540352 +0800","caller":"blockio/pipeline.go:469","msg":"NumericSensor[\"IO/PrefetchDepth:1\"] Current:0 State:Green"}
{"level":"INFO","time":"2023/10/20 14:38:06.540413 +0800","caller":"blockio/pipeline.go:474","msg":"MetaCacheWindow: 0/0 | 0/0, MetaCacheTotal: 0/0 | 0/0"}
{"level":"INFO","time":"2023/10/20 14:38:07.546287 +0800","caller":"dbutils/mem.go:105","msg":"HeapInfo:Alloc:228MB TotalAlloc:991MB Sys:340MB HeapAlloc:228MB HeapSys:247MB HeapIdle:13MB HeapReleased:1MB HeapInuse:233MB"}
panic: runtime error: slice bounds out of range [:-1]

goroutine 25683 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/merge.(*Basic).controlMem(0x107718a00?, {0x1094b0c30?, 0x0?, 0x105ceb274?}, 0x1400bee7ae8?)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/merge/policyBasic.go:190 +0x220
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/merge.(*Basic).Revise(0x1400fea9220, 0x1047b4674?, 0x1400bee7b88?)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/merge/policyBasic.go:138 +0x10c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db/merge.(*MergeExecutor).ExecuteFor(0x1400fea9270, 0x1400d7e0a00, {0x0, 0x0, 0x0}, {0x107d12460, 0x1400fea9220})
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/merge/executor.go:146 +0xb4
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.(*MergeTaskBuilder).trySchedMergeTask(0x1400fd8e660)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/scannerop.go:173 +0x80
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.(*MergeTaskBuilder).onPostTable(...)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/scannerop.go:213
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.(*LoopProcessor).OnPostTable(...)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/catalog/processor.go:64
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.(*dbScanner).onPostTable(0x1400fea91d0, 0x0?)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/scanner.go:134 +0x80
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.(*LoopProcessor).OnPostTable(...)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/catalog/processor.go:64
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.(*DBEntry).RecurLoop(0x1400bee7e78?, {0x107d27340, 0x1400fea91d0})
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/catalog/database.go:607 +0xe0
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.(*Catalog).RecurLoop(0x73f3a54800000000?, {0x107d27340, 0x1400fea91d0})
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/catalog/catalog.go:1114 +0xc8
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/db.(*dbScanner).OnExec(0x1400fea91d0)
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/db/scanner.go:74 +0x258
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/worker.(*heartbeater).Start.func1()
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/tasks/worker/heartbeater.go:76 +0x78
created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tasks/worker.(*heartbeater).Start
        /Users/sudong/Program/MatrixOrigin/matrixone/pkg/vm/engine/tae/tasks/worker/heartbeater.go:67 +0xa4

log:
[server.log](https://github.com/matrixorigin/matrixone/files/13050779/server.log)

**Comment by XuPeng-SH** [comment]
fixed by #12244

**Comment by aressu1985** [comment]
fixed
